docs/modules/math.rst

```diff
@@ -118,3 +118,44 @@ file and create signatures based on those results.
     Returns 0 or 1, it's useful when writing a score based rule.
 
     *Example: math.to_number(SubRule1) \* 60 + math.to_number(SubRule2) \* 20 + math.to_number(SubRule3) \* 70 > 80*
+
+.. c:function:: count(byte/string, offset, size)
+
+    .. versionadded:: 4.2.0
+
+    Returns how often a specific byte or substring occurs, starting at *offset*
+    and looking at the next *size* bytes. When scanning a
+    running process the *offset* argument should be a virtual address within
+    the process address space.
+    *offset* and *size* are optional; if left empty, the complete file is searched.
+
+    *Example: math.count("$[]", 0, 100) >= 5*
```

Comment:
The `count` variant receiving a byte is ok to me, but I have concerns about the variant that accepts a string. The concern is that this is very inefficient way of counting the number of occurrences for a string. I'm aware that YARA doesn't offer a way for counting the number of occurrences for a string in a certain offset range, but if that functionality is required it could be implemented in some other way.

Also `math.count("foo")` is redundant, as this can be already done by other means in a more efficient way. In general I think that this variant encourages the use of this function with whole files, and that's not something I would recommend. Even for functions like `hash.sha256()` we require explicit offset and size.

Reply:
> Also math.count("foo") is redundant, as this can be already done by other means in a more efficient way.

Done, I've deleted it.

> The concern is that this is very inefficient way of counting the number of occurrences for a string. I'm aware that YARA doesn't offer a way for counting the number of occurrences for a string in a certain offset range, but if that functionality is required it could be implemented in some other way.

The main use case we had in mind for this function was counting a short and relatively common string that is only required in specific circumstances, e.g. to check for obfuscation. Since the string is only required in comparatively rare cases, adding it as a normal YARA string would slow down performance globally, whereas with this function it could be evaluated only in specific cases.
Quick and dirty example:
`uint16(0) == 0x253c and filesize < 10KB and math.count("$[", 0, filesize) > 500`

I agree that the current implementation is inefficient. I've reworked it somewhat and I'll gladly take hints or references for further optimization.

Reply:
I think I'm fine with this as long as it doesn't encourage users to use the new feature without thinking twice about the implications. The latest changes are probably enough.

Reply:
@secDre4mer, after @wxsBSD's changes the functions that receive strings can be removed and leave only those that receive a single byte.

Reply:
Alright, I'm not 100% convinced that there's no use case for the string count methods (see my example above), but I can understand that it's better to provide a single interface for string counting rather than adding a functional duplicate that works slightly different and is faster only in very specific cases.

Reply:
I think, with only a single byte, we would rewrite this condition 
```
uint16(0) == 0x253c and filesize < 10KB and math.count("$[", 0, filesize) > 500
```
to
```
uint16(0) == 0x253c and filesize < 10KB and math.count("$", 0, filesize) > 500 and math.count("[", 0, filesize) > 500
```
which is more fuzzy but could still work on obfuscated payloads.